### DISCUSSION TITLE
Why dropping a large table does not change the size of the database?

### DISCUSSION THREAD

**Discussion by vinh-nguyen-phd** [duckdb/duckdb#7152] [General]
I dropped a huge table in my database. However, the size of the database stayed the same. I tried to use commit() and VACCUM statements but nothing changes. What should I do if I want to drop tables to reduce the size of the database? Thanks.

**Comment by Alex-Monahan** [comment]
Hello! Today we do not return that space back to the OS. We will reuse it if you insert new data though, I believe! Fixing this is on the roadmap.

**Reply by vinh-nguyen-phd** [reply]
Thank you for your response. I tested it by adding a large table, dropping it, then adding the same table again. The database keeps getting larger rather than reusing the space. Hope this issue will be fixed in the future updates.

**Reply by Tishj** [reply]
I think you need to checkpoint in between the drop and the re-add, otherwise the system isn't aware that these blocks are free

**Reply by vinh-nguyen-phd** [reply]
What is the syntax to do that? Thank you.

**Reply by Tishj** [reply]
```sql
CHECKPOINT;
```
<https://duckdb.org/docs/sql/statements/checkpoint>

**Reply by vinh-nguyen-phd** [reply]
Oh, thanks so much. I will check it out.

**Reply by bladko-tom** [reply]
@Tishj  i have a question regarding this topic

If data is INSERT OR REPLACE with active primary key (2 columns) and the same set of data is being inserted to DB, then of course number of records will be same but DB is still growing. 

But it looks like index related to primary key is growing, not the storage related to table itself (at based on entire insert size vs only 2 columns on index)

Does my understanding is right ?
